tools/generator/src/XCSchemeEnvrionmentVariables+Extensions.swift

```diff
@@ -0,0 +1,25 @@
+import XcodeProj
+
+extension Array where Element == XCScheme.EnvironmentVariable {
+    /// Provides the Bazel-specific environment variables for Build with Bazel
+    /// launch action.
+    ///
+    /// For more information:
+    /// https://docs.bazel.build/versions/main/user-manual.html#run
+    static let bazelLaunchVariables: [XCScheme.EnvironmentVariable] = [
+        .init(
+            variable: "BUILD_WORKSPACE_DIRECTORY",
+            value: "$(SRCROOT)",
+            enabled: true
+        ),
+        .init(
+            variable: "BUILD_WORKING_DIRECTORY",
+            // This is a poor substitute for the working directory.
+            // Preferably, it would be $(PWD) or something similar.
+            // Unfortunately, none of the following worked:
```

Comment:
We can remove this last line. Xcode doesn't have a concept of a working directory. Schemes can have a working directory though, and they default to something inside of derives data. Maybe we should set this value to whatever that defaults to, or change the working directory to be SRCROOT?

Reply:
I see [a way to specify that you want to set a custom working directory](https://github.com/tuist/XcodeProj/blob/main/Sources/XcodeProj/Scheme/XCScheme%2BLaunchAction.swift#L45). Unfortunately, I don't see a way to set the actual custom working directory. 

When I set it in the Xcode UI, I see the following in the scheme file:

```xml
<LaunchAction
      useCustomWorkingDirectory = "YES"
      customWorkingDirectory = "/Users/chuckgrindel"
```

Do you know if there is a macro for the derives data location? I will poke around to try and find one.

Reply:
It looks like `BUILD_PRODUCTS_DIR` evaluates to the directory that Xcode uses as the current working directory.